src/language/concurrency.md

```diff
@@ -16,90 +19,99 @@ prevpage:
+and `Stream`, and `isolates`, which allow you to move processes to separate
+cores. All Dart code runs in isolates.
+
+An isolate is like a small space on your machine that has its own isolated
+memory. If you’re coming from another language that supports multiple threads,
+you’ll find that isolates are similar, but are crucially different in that
+isolates can’t share memory. Most Dart apps run all their code in a single
+isolate, the main isolate, but you can create more isolates if you need them.
+When you spawn a new isolate, it will have its own isolated memory heap, and it
+will have its own event loop. The event loop is what makes asynchronous and
+concurrent programming possible in Dart.
+
+## Event Loop
+
+Dart’s runtime model is based on an event loop. This event loop is responsible
+for executing the code of your program, collecting and processing events,
+running garbage collection processes, and more. As your application runs, all
+events are added to a queue, called the event queue. Events can be anything from
+requests to repaint the UI, to user taps and keystrokes, to i/o from the disk.
+Because your app can’t predict what order these events will happen, it runs this
+event loop that processes events in the order they're queued, one at a time.
 
-## Asynchrony types and syntax
+![A figure showing events being fed, one by one, into the event loop](/assets/img/language/concurrency/event-loop.png)
 
-If you're already familiar with `Future`, `Stream`, and async-await,
-then you can skip ahead to the [isolates section][].
+Implementing an event loop might resemble this:
 
-[isolates section]: #how-isolates-work
+```dart
+while (eventQueue.waitForEvent()) {
+  eventQueue.processNextEvent();
+}
+```
 
+Even though this event loop is synchronous and runs on a single thread, most
+Dart applications need to do more than one thing at a time. For example, a
+client application might need to execute an HTTP request, while also listening
+for a user to tap a button. To handle this, Dart offers many async APIs, like
+Futures, Streams, and async-await. These APIs are built around this event loop.
 
-### Future and Stream types
+For example, consider making a network request:
+```dart
+http.get('https://example.com').then((response) {
+  if (response.statusCode == 200) {
+    print('Success!')'
+  }  
+}
+```
 
-The Dart language and libraries use `Future` and `Stream` objects to
-represent values to be provided in the future.
-For example, a promise to eventually provide an `int` value
-is typed as `Future<int>`.
-A promise to provide a series of `int` values
-has the type `Stream<int>`.
+When this code is processed by the event loop, it will immediately call the
+first clause - `http.get`, and return a `Future`. It will also tell the event
+loop to take note that sometime in the future, a specific event will happen (
+which is in this case an HTTP response), and when that event happens, it should
+execute the callback passed to `then`.
```

Comment:
```suggestion
When this code reaches the event loop, it immediately calls the
first clause -- `http.get`, and returns a `Future`. It also tells the event
loop to take note that sometime in the future, a specific event will happen
(in this case, an HTTP response), and when that event happens, it should
execute the callback passed to `then`.
```

Questions:
First sentence: Just from looking at the code, it's not obvious why it returns a Future. Is it `then` that causes that to happen? Or is Future the return type of `http.get`? 

Second sentence: it says it "*also* tells the event loop..." to expect a future, but isn't the code telling the event loop to expect a Future *because* the first clause returns a Future, not in addition-to? If that's the case, could change it to something like: "The returned `Future` tells the event loop to look out for a specific event to occur (in this case, ..."

Also, last clause: what's a callback? Why is it passed to `then`? Is the callback the result of the `Future`?